app/src/main/java/com/hwang/xsighting/MainActivity.java

```diff
@@ -43,222 +45,75 @@
-                    });
-          }
-        } else {
-          Log.d(TAG, "get failed with ", task.getException());
-        }
-      }
-    });
-  }
-  
-  private void updateRecyclerView() {
-
-    recyclerView = findViewById(R.id.recyclerview_allsightings);
-//    recyclerView.setHasFixedSize(true);
-
-    // Creates a layout manager and assigns it to the recycler view
-    layoutManager = new LinearLayoutManager(this);
-    recyclerView.setLayoutManager(layoutManager);
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
 
-    // Specifies which adapter the recycler view should use
-    adapter = new AllSightingsAdapter(new ArrayList<Sighting>());
-    recyclerView.setAdapter(adapter);
 
-    FirebaseFirestore db = FirebaseFirestore.getInstance();
-    // Grab sighting and sort by created time
-    db.collection("sighting").orderBy("createdTime", Query.Direction.ASCENDING)
-            .addSnapshotListener(new EventListener<QuerySnapshot>() {
-              @Override
-              public void onEvent(@Nullable QuerySnapshot snapshots,
-                                  @Nullable FirebaseFirestoreException e) {
-                if (e != null) {
-                  Log.w(TAG, "listen:error", e);
-                  return;
-                }
+      loadFragment(new SightingsFragment());
 
-                for (DocumentChange dc : snapshots.getDocumentChanges()) {
-                  switch (dc.getType()) {
-                    case ADDED:
-                      Log.d(TAG, "New sighting: " + dc.getDocument().getData());
-                      adapter.add(dc.getDocument().toObject(Sighting.class), dc.getDocument().getId());
-                      break;
-                    case MODIFIED:
-                      Log.d(TAG, "Modified sighting: " + dc.getDocument().getData());
-                      break;
-                    case REMOVED:
-                      Log.d(TAG, "Removed sighting: " + dc.getDocument().getData());
-                      adapter.remove(dc.getDocument().toObject(Sighting.class));
-                      break;
-                  }
-                }
+      BottomNavigationView bottomNavigationView = (BottomNavigationView)
+              findViewById(R.id.navigation);
+      bottomNavigationView.setOnNavigationItemSelectedListener(mOnNavigationSelectedListener);
 
-              }
-            });
   }
 
-  public void setNavigation(){
-
-    setContentView(R.layout.activity_main);
-    BottomNavigationView bottomNavigationView = (BottomNavigationView)
-            findViewById(R.id.navigation);
-    bottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_LABELED);
-    updateRecyclerView();
-    bottomNavigationView.setOnNavigationItemSelectedListener(
+  //TODO: if we want to use this as a receiver for redirection, then we need to edit the onactivity for createsighting to onactivityresult

```

Comment:
67-89
This is a potential solution, but this would involve us in altering the `onactivityresult()` in our ViewSightingsClass.

The purpose of this is to allow a ViewSightings activity to redirect to MainActivity's CreateSightingFragment